components/backup/src/endpoint.rs

```diff
@@ -1408,31 +1402,38 @@ pub mod tests {
         assert!(endpoint.pool.borrow().size == 3);
     }
 
+    pub struct EndpointWrapper<E: Engine, R: RegionInfoProvider> {
+        inner: Arc<Mutex<Endpoint<E, R>>>,
```

Comment:
Is it necessary for `Arc<Mutex<_>>`? Seems all methods are mutable. And it won't be scheduled to multi threads at one time.

Reply:
Yes. But this is only for test.... This test need to call  the method of `Endpoint` in both test-thread and worker-thread.